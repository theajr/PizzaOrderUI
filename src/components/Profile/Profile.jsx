import React, { useState } from 'react';
import { connect } from '../../util/appContext';

import { makeStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardActionArea from '@material-ui/core/CardActionArea';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';
import AddressForm from '../AddressForm/AddressForm';
import { Collapse } from '@material-ui/core';
import AddressesList from '../AddressesList/AddressesList';
import AddCircleOutlineIcon from '@material-ui/icons/AddCircleOutline';

const useStyles = makeStyles(theme => ({
  card: {
    // maxWidth: 300,
  },
  root: {
    display: 'flex',
    justifyContent: 'space-around',
    [theme.breakpoints.down('md')]: {
      flexDirection: 'column',
    },
  },
}));

const Profile = React.memo(props => {
  const { profile, totalOrders } = props;
  const [addAdress, setAddAddress] = useState(false);

  const classes = useStyles();

  return (
    <div className={classes.root}>
      <Card className={classes.card}>
        <CardMedia
          component="img"
          alt={profile.first_name}
          height="140"
          image={
            profile.avatar || 'https://www.w3schools.com/howto/img_avatar.png'
          }
          title={profile.first_name}
        />
        <CardContent>
          <Typography gutterBottom variant="h5" component="h2">
            {profile.first_name}, {profile.last_name}
          </Typography>
          <Typography variant="caption" color="textSecondary" component="p">
            Orders Placed so far: <b>{totalOrders} </b>
          </Typography>
        </CardContent>
      </Card>
      <div>
        <Button
          variant="contained"
          color="secondary"
          onClick={() => setAddAddress(true)}
          startIcon={<AddCircleOutlineIcon />}
        >
          Add New Address
        </Button>
        {!addAdress && (
          <>
            {' '}
            <Typography variant="h6" gutterBottom align="left">
              Saved Addresses
            </Typography>
            <AddressesList />
          </>
        )}
        {addAdress && <AddressForm onSuccess={() => setAddAddress(false)} />}
      </div>
    </div>
  );
});

const mapStateToProps = state => {
  return {
    totalOrders: state.auth.totalOrders,
    profile: state.auth.profile,
  };
};

const mapDispatchToProps = dispatch => {
  return {
    updateProfile: data => dispatch({ type: 'UPDATE_PROFILE', paylod: data }),
  };
};
export default connect(mapStateToProps, mapDispatchToProps)(Profile);
