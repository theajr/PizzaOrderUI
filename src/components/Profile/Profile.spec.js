import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Profile from './';
import { renderComp } from '../../../tests/utilRender';

describe('<Profile />', () => {
  test('renders', () => {
    const wrapper = renderComp(Profile);
    expect(wrapper).toMatchSnapshot();
  });
});
