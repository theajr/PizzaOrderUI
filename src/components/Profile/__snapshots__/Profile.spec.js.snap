// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> renders 1`] = `ShallowWrapper {}`;
